src/Umbraco.Cms.ManagementApi/Controllers/NewInstallController.cs

```diff
@@ -0,0 +1,131 @@
+using Umbraco.Cms.Core.Configuration.Models;
+using Umbraco.Cms.Core.Hosting;
+using Umbraco.Cms.Core.Install.Models;
+using Umbraco.Cms.Core.Mapping;
+using Umbraco.Cms.Infrastructure.Install;
+using Umbraco.Cms.Infrastructure.Migrations.Install;
+using Umbraco.Cms.ManagementApi.Filters;
+using Umbraco.Cms.ManagementApi.ViewModels.Installer;
+using Umbraco.Extensions;
+using Umbraco.New.Cms.Core.Factories;
+using Umbraco.New.Cms.Core.Models.Installer;
+using Umbraco.New.Cms.Core.Services.Installer;
+using Umbraco.New.Cms.Web.Common.Routing;
+
+namespace Umbraco.Cms.ManagementApi.Controllers;
+
+[ApiController]
+[ApiVersion("1.0")]
+[BackOfficeRoute("api/v{version:apiVersion}/install")]
+[RequireRuntimeLevel(RuntimeLevel.Install)]
+public class NewInstallController : Controller
+{
+    private readonly IUmbracoMapper _mapper;
+    private readonly IInstallSettingsFactory _installSettingsFactory;
+    private readonly IInstallService _installService;
+    private readonly GlobalSettings _globalSettings;
+    private readonly IHostingEnvironment _hostingEnvironment;
+    private readonly InstallHelper _installHelper;
+    private readonly DatabaseBuilder _databaseBuilder;
+
+    public NewInstallController(
+        IUmbracoMapper mapper,
+        IInstallSettingsFactory installSettingsFactory,
+        IInstallService installService,
+        IOptions<GlobalSettings> globalSettings,
+        IHostingEnvironment hostingEnvironment,
+        InstallHelper installHelper,
+        DatabaseBuilder databaseBuilder)
+    {
+        _mapper = mapper;
+        _installSettingsFactory = installSettingsFactory;
+        _installService = installService;
+        _globalSettings = globalSettings.Value;
+        _hostingEnvironment = hostingEnvironment;
+        _installHelper = installHelper;
+        _databaseBuilder = databaseBuilder;
+    }
+
+    [HttpGet("settings")]
+    [MapToApiVersion("1.0")]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status428PreconditionRequired)]
+    [ProducesResponseType(typeof(InstallSettingsViewModel), StatusCodes.Status200OK)]
+    public async Task<ActionResult<InstallSettingsViewModel>> Settings()
+    {
+        // Register that the install has started
+        await _installHelper.SetInstallStatusAsync(false, string.Empty);
+
+        InstallSettingsModel installSettings = _installSettingsFactory.GetInstallSettings();
+        InstallSettingsViewModel viewModel = _mapper.Map<InstallSettingsViewModel>(installSettings)!;
+
+        return viewModel;
+    }
+
+    [HttpPost("setup")]
+    [MapToApiVersion("1.0")]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status428PreconditionRequired)]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    public async Task<IActionResult> Setup(InstallViewModel installData)
```

Comment:
Should it be possible to override a preconfigured DB name with the data in `installData`? 

If I have a preconfigured connection string like this: `"Data Source=|DataDirectory|/ThisIsMyDbName.sqlite.db;Cache=Shared;Foreign Keys=True;Pooling=True"`

...and I use the following `installData`:

```json
{
  "user": {
      "name": "****",
      "email": "****@****.**",
      "password": "****",
      "telemetryLevel": "Basic"
  },
  "database": {
    "Name": "MyNewDb",
    "providerName": "Microsoft.Data.SQLite",
    "id": "530386a2-b219-4d5f-b68c-b965e14c9ac9"
  },
  "subscribeToNewsletter": false
}
```

...I would expect the created DB to be called named "MyNewDb" (from the action payload), but it ends up being called "ThisIsMyDbName" (from the preconfigured connection string).

I believe this is not a newly introduced thing - I suspect this is also the case in the current installer. However... should it be handled here? Or should we simply rely on the UI disabling input fields when a preconfigured connection string is present?

Reply:
I just looked a bit at it, and it is indeed not a newly introduced thing.

It's one of those weird places where frontend and backend is coupled together in the current implementation. Currently the frontend essentially "asks" the `DatabaseConfigureStep` if the view should be displayed, via the View property. 

If a connection string is configured the step tries to validate the schema, if the schema is valid it returns false when asked if the view should be displayed, this skips the database part in the frontend, and you're then unable to change it. 

I think the way the frontend plans to tackle this is to call the `settings` endpoint, which will only return the preconfigured database in the databases list with `isConfigured` set to true, and it will then skip the database setup step, like we currently have.

So the short answer is that I think we should still rely on the UI for this one, so if you preconfigure a database, that's the one you're gonna get.

Reply:
That's fine 😄 we can always fix it later if need be.